src/backend/api/meals.js

```diff
@@ -2,11 +2,160 @@ const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
+// Returns meals // GET api/meals/?
 router.get("/", async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    // knex syntax for selecting things
+    let meals = await knex("meals").select("*");
+    const maxPrice = request.query.maxPrice
+    const availableReservations = request.params.availableReservations
+    const title = request.query.title
+    const createdAfter = request.query.createdAfter
+    const limit = request.query.limit
+    if (maxPrice) {
```

Comment:
Instead of loading all the meals from the database and then apply the filters, you can filter the data in your knex expression. Something like:
knex("meals").select('*').where('price', '<', maxPrice).orWhere( ...